synapse/api/filtering.py

```diff
@@ -444,6 +444,9 @@ def lazy_load_members(self):
     def include_redundant_members(self):
         return self.filter_json.get("include_redundant_members", False)
 
+    def add_room_ids(self, room_ids):
```

Comment:
Filters are currently immutable, and I think that's a good thing for being able to reason about them. One way to do this would be to have a `with_room_ids` method which returns a *new* Filter, having replaced the room list.

Needs a docstring too, please.